pkg/tbtc/signing.go

```diff
@@ -0,0 +1,224 @@
+package tbtc
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"github.com/keep-network/keep-core/pkg/chain"
+	"github.com/keep-network/keep-core/pkg/protocol/group"
+	"github.com/keep-network/keep-core/pkg/tecdsa/retry"
+	"github.com/keep-network/keep-core/pkg/tecdsa/signing"
+	"golang.org/x/exp/slices"
+	"math/big"
+	"math/rand"
+	"sort"
+)
+
+// signingRetryLoop is a struct that encapsulates the signing retry logic.
+type signingRetryLoop struct {
+	signingGroupMemberIndex group.MemberIndex
+	signingGroupOperators   chain.Addresses
+
+	chainConfig *ChainConfig
+
+	attemptCounter    uint
+	attemptStartBlock uint64
+	attemptSeed       int64
+
+	randomRetryCounter uint
+	randomRetrySeed    int64
```

Comment:
I see `randomRetrySeed` is used in `qualifiedOperatorsSet` and `attemptSeed` is used for an additional trim. Can't we use `attemptSeed` everywhere? I see no security reasons for using two separate seeds and I must confess `attemptCounter` vs `randomRetryCounter` and `attemptSeed` vs `randomRetrySeed` was (and still is) the biggest point of confusion when reviewing this code.

Reply:
Basically, same response as in https://github.com/keep-network/keep-core/pull/3260#discussion_r971691910, and the separate seed was just for clear separation of those two random operations. I will change to use `attemptSeed` everywhere.

Reply:
See https://github.com/keep-network/keep-core/pull/3260/commits/aabe50e8f6a50c092282be5c31ccff0de6e0d61e